File: library/src/test/java/org/crispy_fish/datatype/underscore_pairs/SimpleStringUnderscorePairWriterTest.java
package org.crispy_fish.datatype.underscore_pairs;

import org.coner.crispy_fish.datatype.underscore_pairs.SimpleStringUnderscorePairWriter;
import org.junit.*;

import static org.assertj.core.api.Assertions.assertThat;

public class SimpleStringUnderscorePairWriterTest {

    private SimpleStringUnderscorePairWriter writer;

    @Before
    public void setup() {
        writer = new SimpleStringUnderscorePairWriter();
    }

    @Test
    public void whenPairValidKeyWithValidValueItShouldWrite() {
        String actual = writer.pair("key", "value");

        assertThat(actual).isEqualTo("_key_value");
    }

    @Test
    public void whenPairValidKeyWithNullValueItShouldWrite() {
        String actual = writer.pair("key", null);

        assertThat(actual).isEqualTo("_key_");
    }

}

File: library/src/test/kotlin/org/coner/crispyfish/datatype/underscorepairs/SimpleStringUnderscorePairWriterTest.kt
package org.coner.crispyfish.datatype.underscorepairs

import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test

class SimpleStringUnderscorePairWriterTest {

    private lateinit var writer: SimpleStringUnderscorePairWriter

    @Before
    fun setup() {
        writer = SimpleStringUnderscorePairWriter()
    }

    @Test
    fun whenPairValidKeyWithValidValueItShouldWrite() {
        val actual = writer.pair("key", "value")

        assertThat(actual).isEqualTo("_key_value")
    }

    @Test
    fun whenPairValidKeyWithNullValueItShouldWrite() {
        val actual = writer.pair("key", null)

        assertThat(actual).isEqualTo("_key_")
    }

}